integration/integration_test.go

```diff
@@ -2231,10 +2215,30 @@ func trustedClusters(t *testing.T, suite *integrationTestSuite, test trustedClus
 	require.NoError(t, aux.StopAll())
 }
 
-func checkGetClusters(t *testing.T, tun reversetunnel.Server) []reversetunnel.RemoteSite {
-	clusters, err := tun.GetSites()
-	require.NoError(t, err)
-	return clusters
+func waitForClusters(t *testing.T, tun reversetunnel.Server, expected int, timeout time.Duration) {
+	tick := time.NewTicker(1 * time.Second)
+	defer tick.Stop()
+	max := time.NewTimer(timeout)
+	defer max.Stop()
```

Comment:
You've basically re-implemented `require.Eventually` here. I would get rid of the ticker + timer and just use the functionality in the `require` package to accomplish this (it allows you to provide an interval as well as a max timeout).